### PR TITLE
automatically expand required forms

### PR DESCRIPTION
**Pull request by tsiq-jimmyguo**

### CODE REVIEW

File: lib/components/fields/ArrayField.js [hunk 1]
```diff
@@ -425,13 +419,7 @@ var ArrayField = function (_Component) {
...
         }
 
-        if (isDefinedFormWithNonEmptyProperties) {
-          schema = (0, _extends4.default)({}, schema, {
-            items: schema.items[0]
-          });
-
-          delete schema.additionalItems;
-        }
+        // let renderFixedArray handle forms with non-empty properties
```
Comment (tsiq-jimmyguo) [inline lib/components/fields/ArrayField.js]: previously, I modified it so it became a normal array schema. That path does not pre-populate array items. I changed it to just leave it as a fixed array schema, and then add custom logic in `renderFixedArray` to handle forms specifically -- other code remains the same.

File: src/components/fields/ArrayField.js [hunk 2]
```diff
@@ -379,14 +376,7 @@ class ArrayField extends Component {
...
 
-      if (isDefinedFormWithNonEmptyProperties) {
-        schema = {
-          ...schema,
-          items: schema.items[0],
-        };
-
-        delete schema.additionalItems;
-      }
+      // let renderFixedArray handle forms with non-empty properties
```
Comment (tsiq-jimmyguo) [inline src/components/fields/ArrayField.js:379]: previously, I modified it so it became a normal array schema. That path does not pre-populate array items. I changed it to just leave it as a fixed array schema, and then add custom logic in renderFixedArray to handle forms specifically -- other code remains the same.

File: lib/components/fields/ArrayField.js [hunk 3]
```diff
@@ -138,8 +138,7 @@ function DefaultArrayItem(props) {
           style: {
```
Comment (tsiq-jimmyguo) [inline lib/components/fields/ArrayField.js]: ignore this file, just the build output

File: lib/components/fields/ArrayField.js [hunk 4]
```diff
@@ -138,8 +138,7 @@ function DefaultArrayItem(props) {
...
             display: "flex",
             justifyContent: "space-around"
-          }
-        },
+          } },
```
Comment (tsiq-jimmyguo) [inline lib/components/fields/ArrayField.js:141]: ignore this file, just the build output

File: src/components/fields/ArrayField.js [hunk 5]
```diff
@@ -578,10 +568,19 @@ class ArrayField extends Component {
...
-      items = items.concat(new Array(itemSchemas.length - items.length));
+    if (isForm(schema)) {
+      const { minItems = 0 } = schema;
+
+      if (required && isForm(schema) && (!items || items.length < minItems)) {
+        items = items || [];
+        items = items.concat(new Array(minItems - items.length).fill());
+      }
+    } else {
+      if (!items || items.length < itemSchemas.length) {
```
Comment (tsiq-jimmyguo) [inline src/components/fields/ArrayField.js:579]: else block contains untouched original code

**Review by tsiq-jeremy — APPROVED** [review]

**Pull request merged**